src/transpilation/transpiler.ts

```diff
@@ -127,8 +127,11 @@ export function getEmitPathRelativeToOutDir(fileName: string, program: ts.Progra
         emitPathSplits[0] = "lua_modules";
     }
 
-    // Make extension lua
-    emitPathSplits[emitPathSplits.length - 1] = trimExtension(emitPathSplits[emitPathSplits.length - 1]) + ".lua";
+    // Set extension
+    const extension = ((program.getCompilerOptions() as CompilerOptions).extension ?? "lua").trim();
+    const trimmedExtension = extension.startsWith(".") ? extension.substring(1) : extension;
```

Comment:
Should we really strip the leading dot?
Maybe there are cases where it's intended, although i can't think of one.

Reply:
I couldn't come up with a situation where doing it this way could be problematic, but I could see a situation where not specifying the dot would be nicer (ie specifying dot in CLI feels weird to me)